tests/admin_inlines/models.py

```diff
@@ -337,3 +337,59 @@ class Profile(models.Model):
     collection = models.ForeignKey(ProfileCollection, models.SET_NULL, blank=True, null=True)
     first_name = models.CharField(max_length=100)
     last_name = models.CharField(max_length=100)
+
+
+# Models for #32219 and earlier functionality
+# Test combinations with verbose_name and verbose_name_plural
+# in model metas and admin inlines.
+class VerboseParent(models.Model):
+    """
+    Has inlines for its admin class that include a verbose_name
+    (but not a verbose_plural name).
+    """
+    pass
+
+
+class VerbosePluralParent(models.Model):
+    """
+    Has inlines for its admin class that include a verbose_plural_name
+    (but not a verbose name).
+    """
+    pass
+
+
+class MultiVerboseParent(models.Model):
+    """
+    Has inlines for its admin class that include both a verbose_name
+    and a verbose_plural name.
+    """
+    pass
+
+
+class NonVerboseChild(models.Model):
+    verbose_parent = models.ForeignKey(VerboseParent, models.CASCADE, null=True)
+    verbose_plural_parent = models.ForeignKey(VerbosePluralParent, models.CASCADE, null=True)
+    multi_verbose_parent = models.ForeignKey(MultiVerboseParent, models.CASCADE, null=True)
+
+    class Meta:
+        # Does not contain verbose_name or verbose_name_plural.
+        pass
+
+
+class VerboseChild(NonVerboseChild):
+
+    class Meta:
+        verbose_name = 'Meta Verbose Child'
+
+
+class VerbosePluralChild(NonVerboseChild):
+
+    class Meta:
+        verbose_name_plural = 'Meta Plural Children'
+
+
+class MultiVerboseChild(NonVerboseChild):
+
+    class Meta:
+        verbose_name = 'Meta Multi Child'
+        verbose_name_plural = 'Meta Multi Children'
```

Comment:
Do we need so many new models? It should be possible to reuse `Profile`, e.g.
```python
class VerboseNameProfile(Profile):
    class Meta:
        verbose_name = 'Model verbose name'


class VerboseNamePluralProfile(Profile):
    class Meta:
        verbose_name_plural = 'Model verbose name plural'


class VerboseNamePluralProfile(Profile):
    class Meta:
        verbose_name = 'Model verbose name'
        verbose_name_plural = 'Model verbose name plural'
```
Also all tests which work without this patch should be moved to a separate commit, as additional test coverage.

Reply:
I'm not sure what you mean by that. Are you proposing a separate PR with tests for only the existing functionality?

Reply:
Well, we want to test 12 combinations, so using 3 types of Parent and 4 types of Child models to achieve that. (We're not testing the combinations where the inlines have neither a `verbose_name` nor a `verbose_name_plural`.) I don't see the benefit of sub-classing a Parent-type model from another test model.